### PR TITLE
fix(wework): keep model picker above embedded browser

### PR DESCRIPTION
**Pull request by feifei325**
## Summary

- render model selector flyouts and switch-confirmation dialogs in document-level portals with viewport collision handling
- coordinate overlay state with the Electron embedded browser so flyouts can cross the browser pane without clipping or click-through
- preserve switch-and-retry intent across the picker-to-dialog handoff, clearing it only on cancellation
- add desktop E2E coverage for narrow composer layout, browser overlap, interaction blocking, restoration, and automatic retry
- preload sidebar DSH test modules outside the per-test hook timeout found by the full pre-push suite

## Verification

- `pnpm exec vitest run src/components/chat/composer/ModelSelector.test.tsx src/components/chat/ChatInput.test.tsx src/components/ui/hover-card.test.tsx src/features/workspace-tabs/WorkspaceTabStrip.test.tsx` — 158 passed
- `VITEST_MAX_WORKERS=4 pnpm exec vitest run src/components/layout/DesktopSidebar.test.tsx --pool=threads` — 112 passed
- focused ESLint and `pnpm exec tsc -b --pretty false`
- `pnpm ai:verify:electron:build`
- `WEWORK_E2E_BACKGROUND_WINDOW=0 ... pnpm e2e:desktop -- --model-switch-only` — six-way Electron model-switch E2E passed
- repository pre-push gate — ESLint, TypeScript, and full renderer unit suite passed

## Summary by CodeRabbit

* **New Features**
  * Redesigned the desktop model selector with flyout menus that adapt to available screen space.
  * Improved model menus in narrow layouts and alongside the embedded browser with viewport-aware sizing and positioning.
  * Added clearer handling for model selection versus dismissal, preserving queued retries after selection.

* **Bug Fixes**
  * Fixed warning and queue dialogs so they layer correctly and restore browser interaction after dismissal.
  * Improved model-switch interactions, browser overlap behavior, and queued message handling while panes are busy.
  * Ensured follow-up guidance can be delivered reliably during active pane turns.

### CODE REVIEW

File: wework/e2e/desktop/modules/window-attachment-flows.mjs [hunk 1]
```diff
@@ -72,6 +72,63 @@ function desktopWindowLogPath() {
...
+    'waitFor',
+    `${ACTIVE_WORKBENCH_SELECTOR} [data-testid="workspace-browser-native-view"]`,
+    { timeoutMs: DEFAULT_STEP_TIMEOUT_MS }
+  )
+  await control.command('waitFor', '[data-testid="workspace-browser-electron-webview"]', {
+    timeoutMs: DEFAULT_STEP_TIMEOUT_MS,
+  })
+  await control.command('waitFor', '[data-testid="right-workspace-resize-handle"]', {
+    timeoutMs: DEFAULT_STEP_TIMEOUT_MS,
+  })
```
[the remark this answers is not in the document]
Comment (feifei325) [reply wework/e2e/desktop/modules/window-attachment-flows.mjs:126]: Keeping the explicit timeout. The renderer waitFor default is DEFAULT_WAIT_TIMEOUT_MS (5s), while DEFAULT_STEP_TIMEOUT_MS is 10s. These waits cover embedded-browser creation and native-view attachment, so removing the options would halve the timeout and make this CI path more prone to startup flakes.

File: wework/src/components/chat/composer/ModelSelector.test.tsx [hunk 2]
```diff
@@ -259,6 +267,187 @@ describe('ModelSelector desktop layout', () => {
...
+    expect(submenu.style.maxHeight).toBe(
+      'min(var(--radix-popover-content-available-height), calc(100vh - 16px))'
+    )
+  })
+
+  test('uses the Codex flyout viewport contract in a narrow viewport', async () => {
+    Object.defineProperty(window, 'innerWidth', {
+      configurable: true,
+      value: 500,
+    })
```
[the remark this answers is not in the document]
Comment (feifei325) [reply wework/src/components/chat/composer/ModelSelector.test.tsx:330]: No change needed: this describe block already restores window.innerWidth and window.innerHeight in afterEach, which Vitest runs after every test including failures. Adding the same restoration to this test’s finally block would duplicate the existing cleanup contract.

File: wework/src/components/layout/useWorkbenchPaneSession.ts [hunk 3]
```diff
@@ -1829,6 +1829,9 @@ export function useWorkbenchPaneSession({
         const submittedInput = (inputOverride ?? input).trim()
         const currentAttachments = attachmentState.attachments
         const hasCodeComments = codeCommentContexts.length > 0
+        const paneIsBusy = currentRuntimeTask
+          ? (lifecycleStore.getTask(currentRuntimeTask)?.derived.isBusy ?? paneStatus.isBusy)
+          : paneStatus.isBusy
```
[the remark this answers is not in the document]
Comment (feifei325) [reply wework/src/components/layout/useWorkbenchPaneSession.ts]: Fixed in 59ed35823. The guidance helper now reads lifecycle-derived busy state and the two guideWhenBusy call sites explicitly preserve the active-turn decision, so a stale render cannot downgrade guidance to a normal send. Added a regression that captures an idle send callback, transitions the lifecycle to busy, and verifies goal guidance still uses guideRuntimeTask. The relevant 448 tests, ESLint, Prettier, TypeScript, and the full pre-push suite pass.
